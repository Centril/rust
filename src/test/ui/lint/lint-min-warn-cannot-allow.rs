#![deny(warnings)]
//~^ NOTE lint level defined here

#![allow(illegal_floating_point_literal_pattern)]

fn _0() {
    if let 0.0 = 0.0 {}
    //~^ ERROR floating-point types cannot be used in patterns
    //~| NOTE #[deny(illegal_floating_point_literal_pattern)] implied by #[deny(warnings)]
    //~| NOTE #[warn(illegal_floating_point_literal_pattern)] is the minimum lint level
    //~| NOTE the lint level cannot be reduced to `allow`
    //~| WARN this was previously accepted
    //~| WARN hard error
    //~| NOTE for more information, see issue #41620
}

fn main() {}
