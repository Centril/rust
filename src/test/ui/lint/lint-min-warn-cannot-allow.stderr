error: floating-point types cannot be used in patterns
  --> $DIR/lint-min-warn-cannot-allow.rs:7:12
   |
LL |     if let 0.0 = 0.0 {}
   |            ^^^
   |
note: lint level defined here
  --> $DIR/lint-min-warn-cannot-allow.rs:1:9
   |
LL | #![deny(warnings)]
   |         ^^^^^^^^
   = note: #[deny(illegal_floating_point_literal_pattern)] implied by #[deny(warnings)]
   = note: #[warn(illegal_floating_point_literal_pattern)] is the minimum lint level
   = note: the lint level cannot be reduced to `allow`
   = warning: this was previously accepted by the compiler but is being phased out
   = warning: it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

error: aborting due to previous error

